drv/ImmutableList.drv

```diff
@@ -451,10 +451,10 @@ public class IMMUTABLE_LIST KEY_GENERIC extends ABSTRACT_LIST KEY_GENERIC implem
 
 #if KEY_CLASS_Reference
 		while(s-- !=  0) if (a1[s] != a2[s]) return false;
+		return true;
```

Comment:
But this is wrong anyway—if keys are references we must use equals().

Said that, this is a wrong copy-and-paste from ArrayList.drv. I think returning Arrays.equals() if perfectly fine in all cases—the previous code could not assume that the two backing arrays had the same length.

Reply:
`KEY_CLASS_Reference` is the case where we are making identity based collection (that test based on being the exact same object, not just `equals`)
`KEYS_REFERENCE` is the one that covers both Object and the Reference variants.

I think maybe we need a macro, `KEYS_USE_REFERENCE_EQUALITY` or something, that is the same value as `KEY_CLASS_Reference` but makes it clearer we are targeting reference equality semantics exactly.

Reply:
Conclusion: I shouldn't read code on a Saturday when the Friday night had been like yesterday. 🤷🏻‍♂️